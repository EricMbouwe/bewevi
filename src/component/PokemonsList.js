import PropTypes from 'prop-types';
import PokemonShortContainer from '../container/PokemonShortContainer';

const PokemonsList = ({ pokemons }) => (
  <div className="pokemons-list main">
    {pokemons.map(pokemon => (
      <PokemonShortContainer
        key={pokemon.name}
        pokemon={pokemon}
      />
    ))}
  </div>

);

PokemonsList.propTypes = {
  pokemons: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};

export default PokemonsList;
