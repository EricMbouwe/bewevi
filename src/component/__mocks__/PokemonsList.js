import React from 'react';
import PropTypes from 'prop-types';

const PokemonsList = ({ pokemons }) => (
  <>
    <div>
      {JSON.stringify(pokemons)}
    </div>
  </>
);

PokemonsList.propTypes = {
  pokemons: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};

export default PokemonsList;
