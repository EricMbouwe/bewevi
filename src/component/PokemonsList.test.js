import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import PokemonsList from './PokemonsList';

jest.mock('../container/PokemonShortContainer');

const pokemons = [
  {
    name: 'spearow',
    url: 'https://pokeapi.co/api/v2/pokemon/21/',
  },
  {
    name: 'fearow',
    url: 'https://pokeapi.co/api/v2/pokemon/22/',
  },
];

describe('<PokemonsList />', () => {
  it('renders other component', () => {
    render(
      <PokemonsList
        pokemons={pokemons}
      />,
    );
    expect(screen.getByText(/{"name":"spearow","url":"https:\/\/pokeapi.co\/api\/v2\/pokemon\/21\/"}/i)).toBeInTheDocument();
    expect(screen.getByText(/{"name":"fearow","url":"https:\/\/pokeapi.co\/api\/v2\/pokemon\/22\/"}/i)).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <PokemonsList
        pokemons={pokemons}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
