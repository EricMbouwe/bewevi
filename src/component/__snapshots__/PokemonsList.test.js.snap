// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonsList /> renders correctly 1`] = `
<div
  className="pokemons-list main"
>
  {"name":"spearow","url":"https://pokeapi.co/api/v2/pokemon/21/"}
</div>
`;
